src/moonlink/src/table_handler.rs

```diff
@@ -262,48 +350,47 @@ impl TableHandler {
 
                             // Iceberg snapshot LSN requirement is not met, record the required LSN, so later commit will pick up.
                             else {
-                                force_snapshot_lsns.entry(requested_lsn).or_default().push(tx);
+                                table_handler_state.pending_force_snapshot_lsns.entry(requested_lsn).or_default().push(tx);
                             }
                         }
                         // Branch to drop the iceberg table and clear pinned data files from the global object storage cache, only used when the whole table requested to drop.
                         // So we block wait for asynchronous request completion.
                         TableEvent::DropTable => {
                             // Fast-path: no other concurrent events, directly clean up states and ack back.
-                            if !mooncake_snapshot_ongoing && !iceberg_snapshot_ongoing {
+                            if !table_handler_state.mooncake_snapshot_ongoing && !table_handler_state.iceberg_snapshot_ongoing {
                                 drop_table(&mut table, event_sync_sender).await;
                                 return;
                             }
 
                             // Otherwise, leave a drop marker to clean up states later.
-                            drop_table_requested = true;
+                            table_handler_state.start_drop_table();
```

Comment:
Hmm, do you think it's better to call it `request_drop_table`? `start_drop_table` makes me feel "drop table" operation will be started async-ly or sync-ly.